app/frontend/src/features/profile/actions/PendingRequest.tsx

```diff
@@ -0,0 +1,61 @@
+/* has to check whether request is sent or received
+if sent, return a simple disabled "pending request" button
+if recieved, return the response button*/
+import { SetMutationError } from "features/connections/friends";
+import useFriendRequests from "features/connections/friends/useFriendRequests";
+import React from "react";
+
+import PendingFriendRequestSent from "./PendingFriendRequestSent";
+import RespondToFriendRequestProfileButton from "./RespondToFriendRequestProfileButton";
+
+interface PendingRequestProps {
+  userId: number;
+  setMutationError: SetMutationError;
+}
+
+function CheckSentRequests(userId: number) {
+  const friendRequestsSent = useFriendRequests("sent").data;
+  const friendRequestToUser = friendRequestsSent?.find(
+    (e) => e.userId === userId
+  );
+  if (friendRequestToUser !== undefined) {
+    const request = friendRequestToUser;
+    return request;
+  } else {
+    return false;
+  }
+}
+
+function CheckReceivedRequests(userId: number) {
+  const { data } = useFriendRequests("received");
+  const friendRequestsReceived = data;
+  const friendRequestFromUser = friendRequestsReceived?.find(
+    (e) => e.userId === userId
+  );
+  if (friendRequestFromUser !== undefined) {
+    const request = friendRequestFromUser;
+    return request;
+  } else {
+    return false;
+  }
+}
+
+function PendingRequest({ userId, setMutationError }: PendingRequestProps) {
+  if (CheckSentRequests(userId) === false) {
+    const request = CheckReceivedRequests(userId);
+    if (request !== false) {
+      return (
+        <RespondToFriendRequestProfileButton
+          friendRequestId={request.friendRequestId}
+          state={request.state}
+          setMutationError={setMutationError}
+        />
+      );
+    }
+  } else if (CheckSentRequests(userId) !== false) {
+    return <PendingFriendRequestSent />;
+  }
+  return null;
+}
```

Comment:
Since these functions are actually hooks, they can't be used in branching statements (like `if`).  Just put your two hooks at the top and then use branching after to return the components you want (usually with a ternary operator rather than an if statement).

Reply:
(The reason hooks can't be used in branching, is because React expects them to always appear in the same order behind the scenes, or it will get their values confused. [More info here](https://www.netlify.com/blog/2019/03/11/deep-dive-how-do-react-hooks-really-work/).)